### PR TITLE
[inductor] label cpu UTs and test with self.device

### PR DESCRIPTION
**Pull request by Le-Zheng**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #115519

**Summary**
Make UTs work for both cpu and cuda by replacing `cpu` with `self.device`

**Test Plan**
```
python -m pytest test/inductor/test_torchinductor.py 
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Review by jgong5 — APPROVED** [review]

File: test/inductor/test_torchinductor.py [hunk 1]
```diff
@@ -3594,7 +3598,7 @@ def fn(*args):
 
     def test_move_arange(self):
         def fn(x):
-            return torch.arange(len(x), device="cpu").to(x.device) + x
+            return torch.arange(len(x), device=self.device).to(x.device) + x
```
Comment (peterbell10) [inline test/inductor/test_torchinductor.py:3601]: This is clearly part of the test that the tensor is not materialized on CPU.

File: test/inductor/test_torchinductor.py [hunk 2]
```diff
@@ -1275,7 +1275,9 @@ def fn(x):
             return torch.arange(0.1, 8.0001, 1, dtype=x.dtype, device=x.device)
 
         # Test that float arguments are truncated to int when dtype is set explicitly
-        make_arg = functools.partial(make_tensor, device="cpu", requires_grad=False)
+        make_arg = functools.partial(
+            make_tensor, device=self.device, requires_grad=False
```
Comment (peterbell10) [inline test/inductor/test_torchinductor.py:1279]: Every test here constructs the inputs on CPU and `self.common` will move it to the correct device. Might not be ideal but all tests do this.

**Review by peterbell10 — CHANGES_REQUESTED** [review]

File: test/inductor/test_torchinductor.py (hunk 2 above)
Comment (jgong5) [reply test/inductor/test_torchinductor.py:1279]: Oh, I didn't realize the inputs would be placed to CUDA by `self.common` (in `check_model_cuda`). If so, we don't need to change anything here then.

Comment (Le-Zheng) [reply test/inductor/test_torchinductor.py:1279]: Thank you for your comments! I will close this pull request.

**Comment by Le-Zheng** [comment]
Close this PR. No need to make this change.
